crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -111,12 +128,20 @@ pub struct MultiThreadedExecutor {
     unapplied_systems: FixedBitSet,
     /// Setting when true applies deferred system buffers after all systems have run
     apply_final_deferred: bool,
-    /// When set, tells the executor that a thread has panicked.
-    panic_payload: Arc<Mutex<Option<Box<dyn Any + Send>>>>,
     /// When set, stops the executor from running any more systems.
     stop_spawning: bool,
 }
 
+/// References to data required by the executor.
+/// This is copied to each system task so that can invoke the executor when they complete.
+// These all need to outlive 'scope in order to be sent to new tasks,
+// and keeping them all in a struct means we can use lifetime elision.
+#[derive(Copy, Clone)]
+pub struct Context<'scope, 'env: 'scope, 'sys> {
```

Comment:
Does this need to be pub?

```suggestion
struct Context<'scope, 'env: 'scope, 'sys> {
```